modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -743,71 +743,90 @@ private boolean isExpressionFunction(MetadataType metadataType)
         }
     }
 
-    private ValueResolver parseDateTime(Object value, DateTimeType dataType, Object defaultValue)
+
+    private static ValueResolver doParseDate(Object value, Class<?> type)
     {
-        Class<?> type = getType(dataType);
-        if (isExpression(value, parser))
-        {
-            return new TypeSafeExpressionValueResolver((String) value, type);
-        }
 
-        if (type.equals(LocalDateTime.class))
+        if (value instanceof String)
         {
-            Date date = doParseDate(value, DATE_FORMAT, defaultValue);
-            return new StaticValueResolver(LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault()));
+            Object constructedValue = null;
+            DateTime dateTime = getParsedDateTime((String) value);
+
+            if (type.equals(LocalDate.class))
+            {
+                constructedValue = LocalDate.of(dateTime.getYear(), dateTime.getMonthOfYear(), dateTime.getDayOfMonth());
```

Comment:
if that's true then you shouldn't be using this class at all. You're using this same parse method for both Date and DateTime metadata types. It looks like you're consciously dropping the time part of it. Actually, given your response I took a second look at the test cases and it looks like although you do have a time value in your tests, you're never asserting that. Did you mean that or was it an oversight?

In any case, you need a parse method which accounts for the full ISO spec. That means, including the time and timezone